src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -473,44 +473,46 @@ internal static void ImportGlobalSymbols(this FileContentManager file, Compilati
-            var statements = BuildStatements(nodeContent.GetEnumerator(), symbolTracker, diagnostics);
-            symbolTracker.EndScope();
+            var inheritedSymbols = context.Symbols.CurrentDeclarations;
+            context.Symbols.BeginScope(requiredFunctorSupport);
+            var statements = BuildStatements(nodeContent.GetEnumerator(), context, diagnostics);
+            context.Symbols.EndScope();
             return new QsScope(statements, inheritedSymbols);
         }
 
         /// <summary>
-        /// Applies the given build function to the position relative to the tree root and the given symbol tracker 
+        /// Applies the given build function to the position relative to the tree root and the given resolution context 
         /// to get the desired object as well as a list of diagnostics.
         /// Adds the generated diagnostics to the given list of diagnostics, and returns the build object.
-        /// Throws an ArgumentNullException if the given build function, symbolTracker, or diagnostics are null.
         /// </summary>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if the given build function, symbolTracker, or diagnostics are null.
+        /// </exception>
         private static T BuildStatement<T>(FragmentTree.TreeNode node, 
-            Func<QsLocation, SymbolTracker<Position>,Tuple<T, QsCompilerDiagnostic[]>> build, 
-            SymbolTracker<Position> symbolTracker, List<Diagnostic> diagnostics)
+            Func<QsLocation, ResolutionContext<Position>, Tuple<T, QsCompilerDiagnostic[]>> build,
+            ResolutionContext<Position> context, List<Diagnostic> diagnostics)
         {
             if (build == null) throw new ArgumentNullException(nameof(build));
-            if (symbolTracker == null) throw new ArgumentNullException(nameof(symbolTracker));
+            if (context == null) throw new ArgumentNullException(nameof(context));
             if (diagnostics == null) throw new ArgumentNullException(nameof(diagnostics));
 
             var statementPos = node.Fragment.GetRange().Start;
             var location = new QsLocation(DiagnosticTools.AsTuple(node.GetPositionRelativeToRoot()), node.Fragment.HeaderRange);
-            var (statement, messages) = build(location, symbolTracker);
-            diagnostics.AddRange(messages.Select(msg => Diagnostics.Generate(symbolTracker.SourceFile.Value, msg, statementPos)));
+            var (statement, messages) = build(location, context);
+            diagnostics.AddRange(messages.Select(msg => Diagnostics.Generate(context.Symbols.SourceFile.Value, msg, statementPos)));
             return statement;
         }
 
         /// <summary>
         /// If the current tree node of the given iterator is a using-block intro,
-        /// builds the corresponding using-statement updating the given symbolTracker in the process,
+        /// builds the corresponding using-statement updating the given symbol tracker in the process,
```

Comment:
I was conflicted about this because `ResolutionContext` is immutable - so it doesn't really update that per se, but it updates the symbol tracker inside the resolution context.

Reply:
I think mentioning the symbol tracker when there is no argument symbolTracker is more confusing. I tend to think about things as "updateable" when they have any kind of handles that can be updated, not necessarily just the top-level ones.